### PR TITLE
[JENKINS-39150] expose diagnostics across all the channels

### PR DESCRIPTION
**Pull request by kohsuke**
Continuing from https://github.com/jenkinsci/remoting/pull/120

No code change since then, just re-targeting stable-2.x

### CODE REVIEW

**Comment by kohsuke** [comment]
Merging based on :bee: given in #120

File: src/main/java/hudson/remoting/Channel.java [hunk 1]
```diff
@@ -1149,6 +1173,73 @@ public void dumpPerformanceCounters(PrintWriter w) throws IOException {
...
+     * it is request or response that's in flight.
+     *
+     * <dt>Pending incoming calls
+     * <dd>
+     * The reverse of "pending outgoing calls".
+     * Number of RPC calls (e.g., method call through a {@linkplain RemoteInvocationHandler proxy})
+     * that the other side has made to us but not yet returned yet.
+     *
+     * @since 2.26.3
+     */
```
Comment (amuniz) [inline src/main/java/hudson/remoting/Channel.java]: Thanks for the Javadocs @kohsuke 👍

**Comment by oleg-nenashev** [comment]
Well, the Javadoc for the *stable* branch is not ideal. We are not supposed to provide new APIs in the stabilization branch, because it may cause binary incompatibilities  for people upgrading from Jenkins LTS to later weekly releases. I'll fix it later before the release

File: src/main/java/hudson/remoting/Channel.java (hunk 1 above)
Comment (oleg-nenashev) [inline src/main/java/hudson/remoting/Channel.java]: Doe not work in such way

File: src/main/java/hudson/remoting/Channel.java [hunk 2]
```diff
@@ -494,7 +515,8 @@ protected Channel(ChannelBuilder settings, CommandTransport transport) throws IO
 
         transport.setup(this, new CommandReceiver() {
             public void handle(Command cmd) {
-                updateLastHeard();
+                commandsReceived++;
```
Comment (rsandell) [inline src/main/java/hudson/remoting/Channel.java]: Risk of overflow?

**Comment by rsandell** [comment]
Would it be worth adding JMX beans for this?

**Comment by oleg-nenashev** [comment]
https://github.com/jenkinsci/remoting/pull/120#discussion_r84303313 in the original PR has been also addressed improperly. Needs fix

**Pull request merged**